### PR TITLE
Issue #64: Add tests for curl, URL basic-auth, and wget

### LINKED ISSUE #64: [Feature Request / Idea]: Detect curl (and other) "Authorization header" usage
### Suggestion / Feature Request

Curl, wget, and other http tools can be used with authorization headers, like the Basic Authorization header, which contains a base64-encoded username:password combo.

For example:
`curl -H 'Authorization: Basic $(printf username:password|base64)' website.com`
is the same as
`curl -H 'Authorization: Basic dXNlcm5hbWU6cGFzc3dvcmQ=' website.com`

According to https://reqbin.com/req/c-haxm0xgr/curl-basic-auth-example, a user can also use the command
`curl -u username:password website.com`

It would be great if shellclear could detect instances of this, as well as in other programs like wget, for example.

### PR DESCRIPTION
**Pull request by MegaManSec**
cURL: The -u flag and the 'Authorization' header may contain plaintext or base64-encoded secrets.
URL: A URL may contain a login combination.
wget: various flags pass around username and passsword combinations.

Fixes #64 (and more).

### CODE REVIEW

**Comment by MegaManSec** [comment]
Actually it seems these do not work (only now able to test). Is the format wrong?
I was using regex101.com to make these, using the 'golang' flavor, which afaik is sufficiently similar to Rust.

**Comment by kaplanelad** [comment]
Thanks @MegaManSec for your contributing,
I have test for each rule in this folder you can ream more [here](https://github.com/rusty-ferris-club/shellclear/blob/main/CONTRIBUTING.md#new-pattern)

Can you please add the following tests?

**Comment by MegaManSec** [comment]
Should be ok now.

File: shellclear/tests/suites_sensitive_patterns/curl_basic_auth.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  test: "curl --header 'authorization: basic `cat /tmp/secret`' https://priv/"
+  expected: "`cat /tmp/secret`"
+
+- name: detect curl header authorization basic with reading from a file alternative method
+  test: "curl --header 'authorization: basic $(cat /tmp/secret)' https://priv/"
+  expected: "$(cat /tmp/secret)"
+
+- name: detect curl -H authorization with secret environment variable
+  test: "curl -H 'Authorization: Basic $SECRET_VAR' https://priv/"
+  expected: YmFzZTY0OnBhc3N3b3JkCg==
```
Comment (kaplanelad) [inline shellclear/tests/suites_sensitive_patterns/curl_basic_auth.yaml:15]: the expected value is not `$SECRET_VAR`?

**Review by kaplanelad — CHANGES_REQUESTED** [review]
Thanks looks good!

You need to add the *.snap files to the commit as well. i missed it in the contributing guid (added it just now) https://github.com/rusty-ferris-club/shellclear/blob/main/CONTRIBUTING.md#create-test-snapshots

**Review by kaplanelad — APPROVED** [review]
Thank you for contributing to my project! Your contribution is an important part of the project's success, and I am grateful for your support.

**Pull request merged**